ecosystem/sep-0029.md

```diff
@@ -0,0 +1,84 @@
+Users frequently forget to fill in a memo in their deposit transactions.
+Dealing with missing memos may result in lost funds and requires manual actions
+from the custodial support staff.
+
+We introduce a basic convention that allows defining memo requirements on the
+account level.
+
+Although this interoperability standard aims to deal with a missing memo problem
+in the first place, it can be also utilized by a wider class of Stellar-based
+applications to designate additional interaction or compliance prerequisites.
+
+## Abstract
+
+An account owner configures memo requirements by adding a data entry that
+specifies the whether a transaction memo is required when transferring funds to
+the account. Each wallet ensures that a memo is attached before submitting
+the transaction to the network.
+
+## Specification
+
+To set up memo prerequisites for an account, the owner submits a transaction
+with `MANAGE_DATA` operation that adds a data entry with the name
+`config.memo_required` and value `1` (a single character string "1" - ASCII
+code 49 or Unicode U+0031).
+
+When a user submits a transaction containing a `PAYMENT`,
+`PATH_PAYMENT_STRICT_SEND`, `PATH_PAYMENT_STRICT_RECEIVE`, or `MERGE_ACCOUNT`
+operation, the wallet fetches account details from Horizon
+[`/accounts/{account}`](https://www.stellar.org/developers/horizon/reference/endpoints/accounts-single.html)
+endpoint for an address specified in the operation `destination` field.
+
+- If the destination account has `config.memo_required` data entry set, the
+wallet verifies that the transaction `memo` field is set.
+- If `memo` is not set, the wallet shows a notification and prevents the
+transaction from submission.
+- For transactions containing more than one operation, the wallet performs the
+same checks for every operation.
+
+The Horizon server executes the same verification routine server-side when it
+receives the transaction and returns an exception without sending it to the
+network if memo requirements are violated. The implementation of blocking
+transactions with invalid memos on the Horizon level does not eliminate the
+need to verify the requirements on the wallet level.
+
+## Rationale
+
+Multiplexed accounts proposal by David Mazières is more advanced and
+comprehensive solution for the problem described above, but its adoption by the
+whole ecosystem and custodial exchanges will require much more time, as well as
+development efforts. Specifying memo requirements in a form of data entry
+attached to the account introduces a simple way of enforcing transaction
+memo requirements without changing the underlying protocol. 
+
+The name of the data entry follows the namespaces format proposed in SEP-18.
+
+Memo type verification routines are intentionally omitted from this SEP in favor
+of a separate standard with extended definition of more granular control over
+the required memo type via a separate `config.memo_required_type` data entry
```

Comment:
Can you make this less prescriptive of what any future standard should look like?
```suggestion
other attributes of memos via a separate data entries
```

Reply:
Rephrased. In any case, I left this in the rationale section only as an illustrative example.